test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -926,4 +926,95 @@ public void testClobColumnWithNullMetadata() throws Exception {
     expected.add("col2", "null");
     assertEquals(expected, response.getMetadata());
   }
+
+  @Test
+  public void testColumnTypeDate() throws Exception {
+    executeUpdate("create table data(id  integer, col  date)");
+    executeUpdate("insert into data(id, col) values('1001', {d '2004-10-06'})");
```

Comment:
Don't quote integers in SQL. It's not actually valid SQL, so it's not portable. We don't care much about portability, but it's just a bad habit.